tests/functional/adapter/concurrent_transaction/test_concurrent_transaction.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+import threading
+
+from dbt.tests.util import run_dbt
+from dbt.tests.adapter.simple_seed.test_seed import SeedConfigBase
+from tests.functional.adapter.concurrent_transaction.fixtures import *
+
+
+class BaseConcurrentTransaction(SeedConfigBase):
+    def __init__(self):
+        self.query_state = None
+
+    def reset(self):
```

Comment:
Does this get called in `SeedConfigBase` somewhere? Otherwise I think it's only called once, in `def setUp()`, which means it can just be moved there in place of the method call.

Reply:
SeedConfigBase is pretty basic - just a `project_config` definition.

Reply:
Yeah, then I wouldn't bother defining this outside of `def setUp()`. `setUp` already indicates that you're resetting.

Reply:
I doubt that gets much of a benefit. `reset` is more explicit. It's more readable, IMO. I can add a docstring?
```
    def setUp(self, project):
        # Resetting the query_state
        self.query_state = {
            'view_model': 'wait',
            'model_1': 'wait',
        }
```
but not so different from what we have.